src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-form/osd-form.component.ts

```diff
@@ -71,14 +79,25 @@ export class OsdFormComponent extends CdForm implements OnInit {
 
   hasOrchestrator = true;
 
+  simpleDeployment = true;
+
+  deploymentOptions: DeploymentOptions | undefined;
+  deploymentOptionNames: { [key in DeploymentOptionNames]: string };
+  info = {
+    'cost-capacity': 'All the available HDDs are selected',
+    throughput: 'HDDs/SSDs are selected for data devices and SSDs/NVMes for DB/WAL devices',
+    iops: 'All the available NVMes are selected'
+  }
```

Comment:
Aha, more string literals in the TS code... If we don't like putting this into this HTML template, perhaps we should have an "OSD Deployment option" component that encapsulates all these and put this things in an HTML template? It's not a good coding practice with MVC patterns to mix presentation/view (layout, messages, etc) within the code (controller).

Reply:
Used the backend to sent these info to the frontend,

Reply:
In that case, are we adding i18n to the Python code too? Otherwise, this strings won't be translated. I'm not saying that's a bad thing (in fact, I'd love to see many front-end code moved to back-end), but if we want to keep the multi-language supports, all the strings need to be i18n'ed.

Reply:
we are i18n ing in the [html](https://github.com/ceph/ceph/pull/45116/files#diff-1b6efd0f5d10792cc89a319460fef3e5e91b82daeb3921766d872fb5943dec98R43).